docs/performance/sharding.md

```diff
@@ -1,41 +1,132 @@
-This guide assumes you know the basics.
+- Too many shards can degrade search performance and make the cluster unstable.
+  This is referred to as _oversharding_.
 
-If you are looking for an intro to sharding, see {ref}`sharding
-<crate-reference:ddl-sharding>`.
-:::
+- Very large shards can slow down cluster operations and prolong recovery times
+  after failures.
 
-## Optimising for query performance
+## Sizing considerations
 
-(sharding-under-allocation)=
+General principles require careful consideration of cluster
+sizing and architecture.
+Keep the following things in mind when building your sharding strategy.
+Each shard incurs overhead in terms of open files, RAM allocation, and CPU cycles
+for maintenance operations.
+
+### Shard size vs. number of shards
+
+The optimal approach balances shard count with shard size. Individual shards should
+typically contain 10-50 GB of data, being the sweet spot for most
+workloads. In large clusters, this often means fewer shards than total CPU cores,
+as larger shards can still be processed efficiently by multiple CPU cores during
+query execution.
+
+### CPU-to-shard ratio
+
+If most nodes have more shards per table than they have CPUs, the cluster can
+experience performance degradations.
+For example, on clusters with substantial CPU resources (e.g., 8 nodes × 32 CPUs
+= 256 total CPUs), creating 256+ shards per table often proves counterproductive.
+If you don't manually set the number of shards per table, CrateDB will make a
+best guess, based on the assumption that your nodes have two CPUs each.
+
+### 1000 shards per node limit
+
+To avoid _oversharding_, CrateDB by default limits the number of shards per node to
+1_000 as a protection limit. Any operation that would exceed that limit
+leads to an exception.
+For an 8-node cluster, this allows up to 8_000 total shards across all tables.
+Approaching this limit typically indicates a suboptimal sharding strategy rather
+than optimal performance tuning. See also relevant documentation about
+{ref}`table reconfiguration <number-of-shards>` wrt. sharding options.
+
+### Partitions
 
-### Under-allocation is bad
+If you are using {ref}`partitioned tables <crate-reference:partitioned-tables>`,
+note that each partition is clustered into as many shards as you configure
+for the table.
+
+For example, a table with four shards and two partitions will have eight
+shards that can be commonly queried across. But a query that only touches
+one partition will only query across four shards.
+
+How this factors into balancing your shard allocation will depend on the
+types of queries you intend to run.
+
+### Replicas
+
+CrateDB also has replicas of data and this results in additional shards in
+the cluster.
+By default, CrateDB uses the replica setting `0-1` on newly created tables,
+so it will end up with twice the number of shards configured. The more
+replicas you add, the higher is the multiplier (x3, x4, etc.) how you
+compute required capacities.
```

Comment:
@seut commented on this at https://github.com/crate/cratedb-guide/pull/355#discussion_r2393583449 positively, where a statement was removed that falsely claimed replica partitions could be ignored in sizing considerations.

> For balancing reads, replicas are used as well (randomly choosing primaries and replicas).

@matriv added to the same conversation at https://github.com/crate/cratedb-guide/pull/355#discussion_r2410008676:

> I would actually mention and maybe point to the reference guide the use of replicas to not loose data when a node goes down, but also to speed up query throughput (concurrent reads on the same table) since replicas are used as well there.

Reply:
Maybe you could provide a concise refactoring of this paragraph, taking all of this into consideration, @coderabbitai?

Reply:
Thanks, b4d51b6ecd4a adds it.